config_src/drivers/solo_driver/MOM_surface_forcing.F90

```diff
@@ -858,26 +883,40 @@ subroutine wind_forcing_by_data_override(sfc_state, forces, day, G, US, CS)
 
   if (CS%read_gust_2d) then
     call data_override(G%Domain, 'gust', CS%gust, day, scale=US%Pa_to_RLZ_T2)
-    do j=G%jsc,G%jec ; do i=G%isc,G%iec
+    if (associated(forces%tau_mag)) then ; do j=G%jsc,G%jec ; do i=G%isc,G%iec
       forces%tau_mag(i,j) = sqrt(temp_x(i,j)**2 + temp_y(i,j)**2) + CS%gust(i,j)
-      forces%ustar(i,j) = sqrt( forces%tau_mag(i,j) * US%L_to_Z / CS%Rho0 )
+    enddo ; enddo ; endif
+    do j=G%jsc,G%jec ; do i=G%isc,G%iec
+      tau_mag = sqrt(temp_x(i,j)**2 + temp_y(i,j)**2) + CS%gust(i,j)
+      ustar_loc(i,j) = sqrt( tau_mag * US%L_to_Z / CS%Rho0 )
     enddo ; enddo
   else
+    if (associated(forces%tau_mag)) then
+      do j=G%jsc,G%jec ; do i=G%isc,G%iec
+        forces%tau_mag(i,j) = sqrt(temp_x(i,j)**2 + temp_y(i,j)**2) + CS%gust_const
+      ! ustar_loc(i,j) = sqrt( forces%tau_mag(i,j) * US%L_to_Z / CS%Rho0 )
```

Comment:
Was this comment intentional?  Should it be removed?

Reply:
This commented out line was retained from before as a reminder that it would be more efficient to calculate ustar_loc in this mathematically equivalent way, with the following block calculating ustar_loc in an else block to handle the case where forces%tau_mag is not associated, but making this line active would change answers at roundoff.

I suppose, though, that this commented out line could be removed.

Reply:
Seems a little bloaty to me, but if it's intentional then I guess it's fine.